### PR TITLE
Switch to acm ctx before checking S3storeprofiles

### PR DESCRIPTION
**Pull request by shylesh**

### CODE REVIEW

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -3752,7 +3752,7 @@ def deploy(self):
         config.switch_ctx(old_ctx)
         # Only on the active hub enable managedserviceaccount-preview
         acm_version = get_acm_version()
-
+        config.switch_acm_ctx()
```
Comment (petr-balogh) [inline ocs_ci/deployment/deployment.py:3755]: Shouldn't we then return back to old_ctx later on in this function as we do on line 3752?

Comment (shylesh) [reply ocs_ci/deployment/deployment.py:3754]: @petr-balogh , I have rearragend the ctx switches between acm and old ctx.

@prsurve , please let me know if this is ok

Comment (prsurve) [reply ocs_ci/deployment/deployment.py:3754]: yes LGTM

**Review by prsurve — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -3749,10 +3749,9 @@ def deploy(self):
             config.switch_ctx(index)
             logger.info("Creating Resource DataProtectionApplication")
             run_cmd(f"oc create -f {constants.DPA_DISCOVERED_APPS_PATH}")
-        config.switch_ctx(old_ctx)
         # Only on the active hub enable managedserviceaccount-preview
         acm_version = get_acm_version()
-
+        config.switch_acm_ctx()
```
Comment (petr-balogh) [inline ocs_ci/deployment/deployment.py:3754]: This line should be moved above line 3753 as get_acm_version require ACM Context

**Pull request merged**